infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackService.kt

```diff
@@ -83,6 +83,22 @@ class LocationTrackService(
         }
     }
 
+    @Transactional
+    fun updateState(id: IntId<LocationTrack>, state: LayoutState): DaoResponse<LocationTrack> {
```

Comment:
Helpottamaan oikea-oppista deleteä. Tämä on vähän niheää koska meillä enimmäkseen vain tallennetaan draft mutta nyt nuo linkkien putsailut edellyttää taikatemppuja. Ehkä ne pitäisi overridettää mukaan normaaliin tallennukseen, mutta sekin on ärsyttävä toteuttaa. En halunnut tässä kohtaa lähteä sitä mylläämään.

Reply:
Nimeäisin funkkarin pidemmäksi (updateLayoutState), mutta ehkäpä tämä olisi vain oma preferenssi. LocationTrackilla on kuitenkin melkoisesti muutakin tilatietoa nimittäin, vaikkei geneeristä state-termiä olekaan kai muualla käytetty.

Reply:
Kaikilla layoutin käsitteillä on ainakin nykyisellään yksikäsitteinen elinkaaren tilakone, jonka tila tuossa on kyseessä. En oikein näe sekaantumisen vaaraa. Pitäisin tämän mielummin 1=1 kentän nimen kanssa. Kaikki bitit on toki tilaa, joten state on kategorisesti vähän huono sana, mutta se on terminä myös ratkon tilakoneen 1=1 vastaava eikä sille ole helppo keksiä oikein parempaakaan korvaaja.

Sinällään tuo nimi LayoutState on itsessään minusta vähän huono tuolle enum-luokalle, mutta se on jo vähän oma remppansa.